sw/host/provisioning/ft_lib/src/lib.rs

```diff
@@ -456,6 +488,67 @@ fn parse_and_endorse_x509_cert(tbs: Vec<u8>, ca_sk: &SecretKey<NistP256>) -> Res
     generate_certificate_from_tbs(tbs, &signature)
 }
 
+fn parse_and_endorse_x509_cert_ckms(tbs: Vec<u8>, ckms_key_id: &str) -> Result<Vec<u8>> {
+    // Let openssl hash and sign the TBS.
+    let base_name = tmpfilename("cert_signing");
+    let binding_tbs = base_name.to_owned() + ".tbs";
+    let binding_sig = base_name.to_owned() + ".sig";
+    let tbs_filename = binding_tbs.as_str();
+    let sig_filename = binding_sig.as_str();
+
+    // Save TBS in a file.
+    let mut file = OpenOptions::new()
+        .write(true)
+        .truncate(true)
+        .create(true)
+        .open(tbs_filename)
+        .context("failed to open tbs file")?;
+    file.write_all(&tbs)?;
+    drop(file);
+
+    let binding_key = String::from("pkcs11:object=") + ckms_key_id;
+    openssl_command(
+        0,
+        &[
+            "dgst",
+            "-sha256",
+            "-engine",
+            "pkcs11",
+            "-keyform",
+            "engine",
+            "-sign",
+            binding_key.as_str(),
+            "-out",
+            sig_filename,
+            tbs_filename,
+        ],
+    )
+    .context("openssl failed to sign certificate digest")?;
+
+    // Read the signature represented as an ASN.1 object.
+    file = OpenOptions::new().read(true).open(sig_filename)?;
+    let mut asn1_sig = Vec::new();
+    file.read_to_end(&mut asn1_sig)?;
+    drop(file);
+
+    // Parse the ASN.1 string into signature components.
+    let ecdsa_sig =
+        EcdsaSig::from_der(&asn1_sig).context("cannot extract ECDSA signature from blob")?;
```

Comment:
I think the `from_bytes` [method](https://docs.rs/ecdsa/latest/ecdsa/der/struct.Signature.html#method.from_bytes) on the `Signature` struct in the `p256` crate should do the same? Then you could avoid an extra crate include (and just update `use p256::ecdsa::SigningKey;` to `use p256::ecdsa::{Signature, SigningKey};` above. (Probably a cleanup that can done in a follow up later too).

Reply:
sg.